internal/handlers/device.go

```diff
@@ -448,3 +458,20 @@ func (api *API) DeleteDevice(c *gin.Context) {
 
 	c.JSON(http.StatusOK, device)
 }
+
+func childPrefixEquals(existingPrefix, newPrefix []string) bool {
+	if len(existingPrefix) != len(newPrefix) {
+		return false
+	}
+	countMap := make(map[string]int)
+	for _, value := range existingPrefix {
+		countMap[value]++
+	}
+	for _, value := range newPrefix {
+		countMap[value]--
+		if countMap[value] < 0 {
+			return false
+		}
+	}
+	return true
```

Comment:
This looks correct to me, but I really wish there was a first-class implementation of Sets for go. What we really want here is to view each set of prefixes as a set and they are equal if the difference between the sets is an empty set.

We could use this library if we wanted to turn the childPrefix storage into a set natively. It would handle things like detecting if the same thing is specified twice as input, it only ends up in the set once. It would also simplify all of the operations where you're having to create maps on the fly to help with comparisons.

https://github.com/deckarep/golang-set